fork-off/src/fetching.rs

```diff
@@ -37,10 +37,22 @@ impl StateFetcher {
                 .await
                 .unwrap();
 
+            let child_storage_map_res = self
+                .client
+                .get_child_storage_for_key(key.clone(), &block)
+                .await;
+
             let mut output_guard = output.lock();
-            output_guard.insert(key, value);
-            if output_guard.len() % LOG_PROGRESS_FREQUENCY == 0 {
-                info!("Fetched {} values", output_guard.len());
+            output_guard.top.insert(key.clone(), value);
+            if let Ok(child_storage_map) = child_storage_map_res {
+                info!("Fetched child trie with {} keys", child_storage_map.len());
+                output_guard
+                    .child_storage
+                    .insert(key.without_child_storage_prefix(), child_storage_map);
```

Comment:
We can omit these if `child_storage_map` is empty.

Reply:
Here success of the rpc means that this is indeed a key which is used to store a child trie, so we probably want to insert that anyway.

Reply:
I see, so we would insert empy storage which is an information as well.